src/java.base/share/classes/java/lang/classfile/attribute/ModuleAttribute.java

```diff
@@ -300,6 +307,7 @@ default ModuleAttributeBuilder opens(PackageDesc pkge, Collection<AccessFlag> op
          * @param service the service class provided
          * @param implClasses the implementation classes
          * @return this builder
+         * @throws IllegalArgumentException if {@code service} or any of the {@code implClasses} represents a primitive type
```

Comment:
Javadoc tip (not that you should use that here) - if the thrown clause becomes too complex, you can also split it into multiple `throws IllegalArgumentException`. Sometimes that can improve readability.

Reply:
Thanks!